src/fsharp/TypeChecker.fs

```diff
@@ -8713,7 +8713,7 @@ and TcMethodApplication
 
     let denv = env.DisplayEnv
 
-    let isSimpleFormalArg (isParamArrayArg, isOutArg, optArgInfo: OptionalArgInfo, _reflArgInfo: ReflectedArgInfo) = 
+    let isSimpleFormalArg (isParamArrayArg, isOutArg, optArgInfo: OptionalArgInfo, _callerInfoInfo: CallerInfoInfo, _reflArgInfo: ReflectedArgInfo) = 
         not isParamArrayArg && not isOutArg && not optArgInfo.IsOptional 
```

Comment:
I suspect you may want to check that callerInfo is NoCallerInfo here.  I believe this relates to the question of what happens on first-class uses of methods with these attributes, see the various uses of ``isSimpleFormalArg`` below